core/src/blockchain/blockchain.rs

```diff
@@ -218,7 +218,7 @@ impl BlockChain {
         let grandparent_hash_of_new_block = parent_details_of_new_block.parent;
         let prev_best_hash = self.best_block_hash();
 
-        if parent_details_of_new_block.total_score + new_header.score() > self.best_proposal_block_detail().total_score
+        if parent_details_of_new_block.number + new_header.number() > self.best_proposal_block_detail().number
```

Comment:
This change is weird because the `parent_details_of_new_block.number + new_header.number() > self.best_proposal_block_detail().number` is not the necessary condition of the best block change. I think you can remove the check.

Reply:
@sgkim126 [What you said](https://github.com/CodeChain-io/foundry/pull/162#issuecomment-586531723) seems to be the case here. @HoOngEe says that this condition is unnecessary, so should I add a condition which is a comparison of `view`? Or should I get rid of all the conditions?

Reply:
To add a condition(comparison of `view`), It seems I have to add `view` to `BlockDetails` and implement calculating the `view` from `seal`.

Reply:
@somniumism I just naively thought the clause is removable because it will not affect the behavior. As @sgkim126 said, the logic implies checking the validity of the height and view by the score, the struct may need `view` field.

Reply:
@somniumism The below change is also the case. Let's keep the logic equivalent.